web/react/components/edit_channel_header_modal.jsx

```diff
@@ -27,27 +31,39 @@ export default class EditChannelHeaderModal extends React.Component {
         }
     }
 
+    componentWillReceiveProps(nextProps) {
+        if (this.state.originalHeader !== nextProps.channel.header) {
```

Comment:
Why not just set the state header if `this.props.channel.header !== nextProps.channel.header`? Having `originalHeader` in the state is kind of redundant (and a misuse of state) when it should already be in the props. If this is an issue with channel being a reference then it should be copied somewhere further up or have the props split up into primitives